umbraco/Umbraco-CMS
# V14: Member collection view/filter endpoint

## Details
- Adding the basic functionality for a members' collection view - additional improvements are to be done in the future, so we can get the FE going with the members section;
  - Members' UI will be aligned with users in Bellisima - therefore, we are creating a filter endpoint (to narrow down search results) rather than a "collection" endpoint as seen for content and media.

Functionality:
- Can get all members regardless of member type by default;
- Can filter by member type (id);
- Can sort by `name`, `email`, `username` and last edited date (`updateDate`), create date (`createDate`) and last login date `lastLoginDate `;
- Can do a textual search using the `filter` parameter - on the name, username, email, or Guid key.

## Test
- The easiest would be to set up everything on v13 DB from a v13 template project and copy over the DB to this v14 branch;
- Create a couple of member types in the Settings section;
- Create a couple of member groups in the Members section;
- Create a few members from the different member types and part of different member groups;
- _**Optional** (login with members to update their `lastLoginDate` value)_:
  - Create Login.cshtml and LoginStatus.cshtml Partial View Macro Files (from snippets);
  - Enable "Use in rich text editor and the grid" in their respective Macros;
  - On a doc type, add RTE property and render it in the Razor view;
  - Create a content node from the doc type with RTE;
  - Add the 2 macros;
  - Login with a couple of members.
- Verify that everything in the Functionality list above works as expected.
